Incorrect codegen for `Deref`/`DerefMut` of sequence

A sequence internally inherits from `std::int64`.

```edgeql
scalar type UserId extending sequence;
```

`gelx` however generates `Deref`/`DerefMut` traits with an `i32` target:

```rust
impl ::std::ops::Deref for UserId {
    type Target = i32;
    fn deref(&self) -> &Self::Target {
        &self.0
    }
}
impl ::std::ops::DerefMut for UserId {
    fn deref_mut(&mut self) -> &mut Self::Target {
        &mut self.0
    }
}
```

This throws the following error during compilation:
```
error[E0308]: mismatched types
   --> src/db/default.rs:415:9
    |
414 |     fn deref_mut(&mut self) -> &mut Self::Target {
    |                                ----------------- expected `&mut i32` because of return type
415 |         &mut self.0
    |         ^^^^^^^^^^^ expected `&mut i32`, found `&mut i64`
    |
    = note: expected mutable reference `&mut i32`
               found mutable reference `&mut i64`
```

Root cause of the issue seems to a hard coded target type `generate.rs` in https://github.com/ifiokjr/gelx/blob/81d2fbb7e3d89101f9858ed79e745d439e991dd0/crates/gelx_core/src/codegen/generate.rs#L174

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -72,10 +72,10 @@ gelx_macros = { path = "./crates/gelx_macros", version = "0.8.2" }
 
 [workspace.metadata.bin]
 cargo-all-features = { version = "1.10.0", bins = ["cargo-build-all-features", "cargo-check-all-features", "cargo-test-all-features"] }
-cargo-deny = { version = "0.18.2" }
+cargo-deny = { version = "0.18.3" }
 cargo-insta = { version = "1.43.1" }
 cargo-llvm-cov = { version = "0.6.16" }
-cargo-nextest = { version = "0.9.98" }
+cargo-nextest = { version = "0.9.99" }
 
 [workspace.lints.rust]
 # Group lints

--- a/crates/gelx_core/src/codegen/generate.rs
+++ b/crates/gelx_core/src/codegen/generate.rs
@@ -171,7 +171,7 @@ pub(crate) fn generate_scalar(
 			}
 		}
 		impl ::std::ops::Deref for #struct_name {
-			type Target = i32;
+			type Target = #wrapped_struct_type;
 			fn deref(&self) -> &Self::Target {
 				&self.0
 			}

--- a/dbschema/default.gel
+++ b/dbschema/default.gel
@@ -11,6 +11,8 @@ module default {
 		constraint max_value(1_000_000);
 	}
 
+	scalar type UserId extending sequence;
+
 	type Simple {
 		required position: Position;
 	}

--- a/dbschema/migrations/00001-m1wiaho.edgeql
+++ b/dbschema/migrations/00001-m1wiaho.edgeql
@@ -1,4 +1,4 @@
-CREATE MIGRATION m1gd2vrayu62apz6e4sv7yxgpubfuh2kbrxq75rf56blckov5atcsa
+CREATE MIGRATION m1wiahotdsp5yuxymm7qfxct4gqhmjpsr7voj3v5rxxke5hc57nxra
     ONTO initial
 {
   CREATE EXTENSION postgis VERSION '3.5';
@@ -12,6 +12,7 @@ CREATE MIGRATION m1gd2vrayu62apz6e4sv7yxgpubfuh2kbrxq75rf56blckov5atcsa
   };
   CREATE SCALAR TYPE default::RelationshipType EXTENDING enum<Follow, Block, Mute>;
   CREATE SCALAR TYPE default::Role EXTENDING enum<None, Editor, Moderator, Admin, Owner>;
+  CREATE SCALAR TYPE default::UserId EXTENDING std::sequence;
   CREATE GLOBAL additional::alternative -> std::str;
   CREATE GLOBAL default::current_user_id -> std::uuid;
   CREATE ABSTRACT TYPE default::CreatedAt {

--- a/devenv.lock
+++ b/devenv.lock
@@ -3,10 +3,10 @@
     "devenv": {
       "locked": {
         "dir": "src/modules",
-        "lastModified": 1749416335,
+        "lastModified": 1750529628,
         "owner": "cachix",
         "repo": "devenv",
-        "rev": "e8fffc7d61259f27c27b0fbe39a052a604807a2e",
+        "rev": "cee0466541d357356b8c1ee0a61f3e0b94c7a54e",
         "type": "github"
       },
       "original": {
@@ -40,10 +40,10 @@
         ]
       },
       "locked": {
-        "lastModified": 1747372754,
+        "lastModified": 1749636823,
         "owner": "cachix",
         "repo": "git-hooks.nix",
-        "rev": "80479b6ec16fefd9c1db3ea13aeb038c60530f46",
+        "rev": "623c56286de5a3193aa38891a6991b28f9bab056",
         "type": "github"
       },
       "original": {
@@ -74,10 +74,10 @@
     },
     "nixpkgs": {
       "locked": {
-        "lastModified": 1749482418,
+        "lastModified": 1750573865,
         "owner": "NixOS",
         "repo": "nixpkgs",
-        "rev": "971969fb3e62adf72fc99eccdd15459d05188415",
+        "rev": "475a00a22d452d2ab0feaf085749a4a9aa5e2576",
         "type": "github"
       },
       "original": {
@@ -89,10 +89,10 @@
     },
     "nixpkgs_2": {
       "locked": {
-        "lastModified": 1749401433,
+        "lastModified": 1750386251,
         "owner": "NixOS",
         "repo": "nixpkgs",
-        "rev": "08fcb0dcb59df0344652b38ea6326a2d8271baff",
+        "rev": "076e8c6678d8c54204abcb4b1b14c366835a58bb",
         "type": "github"
       },
       "original": {
@@ -118,10 +118,10 @@
         "nixpkgs": "nixpkgs_2"
       },
       "locked": {
-        "lastModified": 1749436897,
+        "lastModified": 1750560265,
         "owner": "oxalica",
         "repo": "rust-overlay",
-        "rev": "e7876c387e35dc834838aff254d8e74cf5bd4f19",
+        "rev": "076fdb0d45a9de3f379a626f51a62c78afe7efb1",
         "type": "github"
       },
       "original": {

--- a/examples/gelx_example/src/db/default.rs
+++ b/examples/gelx_example/src/db/default.rs
@@ -65,8 +65,8 @@ impl __g::gel_protocol::queryable::Queryable for Position {
             ctx,
             type_pos,
             __g::uuid::Uuid::from_bytes([
-                198u8, 101u8, 10u8, 120u8, 66u8, 209u8, 17u8, 240u8, 175u8, 163u8, 31u8,
-                171u8, 222u8, 52u8, 143u8, 173u8,
+                89u8, 58u8, 3u8, 164u8, 79u8, 56u8, 17u8, 240u8, 176u8, 232u8, 45u8,
+                99u8, 243u8, 120u8, 67u8, 128u8,
             ]),
             "default::Position",
         )?;
@@ -171,6 +171,64 @@ mod team {
 }
 mod user {
     use super::*;
+}
+#[derive(::std::fmt::Debug, ::core::clone::Clone)]
+#[cfg_attr(
+    feature = "with_serde",
+    derive(__g::serde::Serialize, __g::serde::Deserialize)
+)]
+pub struct UserId(pub i64);
+#[cfg(feature = "with_query")]
+impl __g::gel_protocol::queryable::Queryable for UserId {
+    type Args = <i64 as __g::gel_protocol::queryable::Queryable>::Args;
+    fn decode(
+        decoder: &__g::gel_protocol::queryable::Decoder,
+        args: &Self::Args,
+        buf: &[u8],
+    ) -> Result<Self, __g::gel_protocol::errors::DecodeError> {
+        Ok(Self(i64::decode(decoder, args, buf)?))
+    }
+    fn check_descriptor(
+        ctx: &__g::gel_protocol::queryable::DescriptorContext,
+        type_pos: __g::gel_protocol::descriptors::TypePos,
+    ) -> Result<Self::Args, __g::gel_protocol::queryable::DescriptorMismatch> {
+        __g::check_scalar(
+            ctx,
+            type_pos,
+            __g::uuid::Uuid::from_bytes([
+                89u8, 59u8, 83u8, 48u8, 79u8, 56u8, 17u8, 240u8, 154u8, 10u8, 101u8,
+                93u8, 244u8, 5u8, 172u8, 58u8,
+            ]),
+            "default::UserId",
+        )?;
+        Ok(())
+    }
+}
+impl ::core::convert::From<UserId> for __g::gel_protocol::value::Value {
+    fn from(value: UserId) -> Self {
+        value.0.into()
+    }
+}
+impl ::core::convert::From<UserId> for i64 {
+    fn from(value: UserId) -> Self {
+        value.0
+    }
+}
+impl ::core::convert::From<i64> for UserId {
+    fn from(value: i64) -> Self {
+        UserId(value)
+    }
+}
+impl ::std::ops::Deref for UserId {
+    type Target = i64;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+impl ::std::ops::DerefMut for UserId {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
 }
 mod wallet {
     use super::*;

